src/ai/backend/manager/api/session.py

```diff
@@ -2179,6 +2215,52 @@ async def shutdown_service(request: web.Request, params: Any) -> web.Response:
     return web.Response(status=204)
 
 
+async def find_dependent_sessions(
+    root_session_name_or_id: uuid.UUID | str,
+    db_connection: SAConnection,
+    access_key: AccessKey,
+) -> Set[uuid.UUID | str]:
+    async def _find_dependent_sessions(session_name_or_id: uuid.UUID | str):
+        session_infos = await match_session_ids(
+            session_name_or_id,
+            access_key=access_key,
+            db_connection=db_connection,
+        )
+
+        assert len(session_infos) >= 1, "session not found!"
+
+        session_id = str(session_infos[0].get("session_id"))
+
+        dependent_session_getter_query = (
+            sa.select(
+                [
+                    session_dependencies.c.session_id,
+                ]
+            )
+            .select_from(session_dependencies)
+            .where(session_dependencies.c.depends_on.in_([session_id]))
+        )
+
+        dependent_session_ids = set(
+            [
+                dependent_session["session_id"]
+                for dependent_session in await db_connection.execute(dependent_session_getter_query)
+            ]
+        )
```

Comment:
This code could be refactored like using the set-comprehension syntax:
```suggestion
        dependent_session_ids = {
            dependent_session["session_id"]
            for dependent_session in await db_connection.execute(dependent_session_getter_query)
        }
```